### PR TITLE
Add WebDriverIO test setup - RSC-158

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-158

This is the first step to getting Applitools tests for RSC.  This PR adds a functional test setup using Webdriver IO, using a docker container for the Chrome similar to the IQ build. One simple "test" that takes a local screenshot.  Locally I have confirmed that the resulting screenshot shows the loaded page.

Note: the current version of WebdriverIO requires node.js 12

### CODE REVIEW

File: gallery/wdio.conf.js [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    /**
+     * Gets executed before a worker process is spawned and can be used to initialise specific service
+     * for that worker as well as modify runtime environments in an async fashion.
+     * @param  {String} cid      capability id (e.g 0-0)
+     * @param  {[type]} caps     object containing capabilities for session that will be spawn in the worker
+     * @param  {[type]} specs    specs to be run in the worker process
+     * @param  {[type]} args     object that will be merged with the main configuration once worker is initialised
+     * @param  {[type]} execArgv list of string arguments passed to the worker process
+     */
+    // onWorkerStart: function (cid, caps, specs, args, execArgv) {
```
Comment (raveneyex) [inline gallery/wdio.conf.js:185]: I assume that you copied this config file from a template and that all of these hooks that aren't used right now will be gone on cleanup, right?

Comment (rpokorny) [reply gallery/wdio.conf.js:185]: > I assume that you copied this config file from a template and that all of these hooks that aren't used right now will be gone on cleanup, right?

Well, generated by an init command, but close enough.  I wasn't planning to remove all the commented out stuff.  Do you think I should?

File: Jenkinsfile [hunk 2]
```diff
@@ -6,9 +6,14 @@
...
 @Library(['private-pipeline-library', 'jenkins-shared']) _
 
+def seleniumDockerImage = 'selenium/standalone-chrome'
+def seleniumDockerVersion = '3.141.59-20200409'
+
 dockerizedBuildPipeline(
   prepare: {
     githubStatusUpdate('pending')
+
+    sh "docker run --name selenium-chrome -d -p 4444:4444 $seleniumDockerImage:$seleniumDockerVersion"
```
Comment (tneer) [inline Jenkinsfile:16]: Minor, but I generally prefer the curly brace form of groovy string interpolation.  Especially when using the `sh` call.  

https://docs.sonatype.com/display/CDI/Jenkins+Pipeline+Shell+Script+Gotcha%27s#JenkinsPipelineShellScriptGotcha's-recRecommendations

File: Jenkinsfile [hunk 3]
```diff
@@ -100,5 +108,11 @@ dockerizedBuildPipeline(
   },
   onFailure: {
     githubStatusUpdate('failure')
+  },
+  cleanup: {
+    sh """
+      docker rm -f selenium-chrome
+      docker rmi $seleniumDockerImage:$seleniumDockerVersion
```
Comment (tneer) [inline Jenkinsfile:115]: Another place that _could_ use curly braces.

File: gallery/wdio.conf.js (hunk 1 above)
Comment (raveneyex) [reply gallery/wdio.conf.js:185]: > > I assume that you copied this config file from a template and that all of these hooks that aren't used right now will be gone on cleanup, right?
> 
> Well, generated by an init command, but close enough. I wasn't planning to remove all the commented out stuff. Do you think I should?

I don't feel strongly about it, though it does improve readability a bit. 
I will ultimately leave the decision to you.

**Pull request merged**